iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -593,6 +594,9 @@ private Path getColStatsPath(Table table) {
   }
 
   private Path getColStatsPath(Table table, long snapshotId) {
+    if (((BaseTable) table).operations() instanceof HadoopTableOperations) {
+      return new Path(table.location() + STATS + snapshotId);
```

Comment:
I am not sure this path is reasonable. Please provide your suggestion here. Thanks.

Reply:
AFAIK that is supposed to be the puffin stats file path for the provided snapshot

Reply:
Yes, this path is puffin stats file path. Hadoop tables and Hive-catalog tables are really no different, they both can use the stats to optimize query. After this PR, the puffin stats file for Hadoop tables is created correctly, and query can get stats from the puffin stats file. So i think we definitely need the stats file for Hadoop tables.

What i want to say, should we keep the same path format for both Hadoop tables and hive-catalog tables? Then we don't need to judge this table type.
 e.g.
```
  private Path getColStatsPath(Table table, long snapshotId) {
+    return new Path(table.location() + STATS + snapshotId);
-    return new Path(table.location() + STATS + table.name() + snapshotId);
  } 
```
But I'm not sure if there will be backward compatibility issues for hive-catalog tables.

Reply:
makes sense if we use stats for Hadoop tables, I wasn't sure about that, most metadata optimizations are available when we use HiveCatalog.

but anyways, not sure why we needed table name in stats file, maybe we should use similar to manifestList naming:
"snap-%d-%s", snapshotId(), UUID.randomUUID().toString())

Reply:
We don't need table name in stats file as the parent path already has table name which can be used to identify the specific table.
Use the `snap` as prefix is reasonable for me.  And i think there is no compatibility issues if changing the hive-catalog colpath to keep the same path format as HadopTable. Changed in https://github.com/apache/hive/pull/4871/commits/b1ffb414699b1cf6778b10b7f74bbbdc02a5d6cd

Reply:
👍